internal/subnamespace/sync.go

```diff
@@ -394,8 +395,8 @@ func (r *SubnamespaceReconciler) enqueueChildrenSNSToRPConversionEvents(snsObjec
 
 // IsUpdateNeeded gets a subnamespace object, a []danav1.Namespaces and two resource lists and returns whether
 // the subnamespace object status has to be updated.
-func IsUpdateNeeded(sns client.Object, childrenRequests []danav1.Namespaces, allocated, free corev1.ResourceList) bool {
-	if !NamespacesEqual(sns.(*danav1.Subnamespace).Status.Namespaces, childrenRequests) ||
+func IsUpdateNeeded(sns client.Object, childrenRequests []danav1.Namespaces, allocated, free corev1.ResourceList, snsContext context.Context, snsClient client.Client) bool {
```

Comment:
Call it `ctx` and `k8sClient` instead of `snsContext` and `snsClient`. Also, move these parameters to be the first ones in the function signature, `context` usually goes first in a signature